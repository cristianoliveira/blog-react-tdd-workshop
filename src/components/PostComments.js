import React from 'react';
import PropTypes from 'prop-types';

const Comment = ({body}) => <div className="post-comment">{body}</div>;

function PostComments({comments, onCommentFormSubmit, onFocus}) {
  if (!comments.length) {
    return null;
  }

  return (
    <div>
      <div className="post-comments">
        {comments.map((comment, i) => <Comment key={i} {...comment} />)}
      </div>
      <form
        className="post-form-comment"
        onSubmit={onCommentFormSubmit}>
        <label>New commentary</label>
        <textarea
          onFocus={onFocus}
          name="text"
          id="text"
          rows="8"
          cols="40"
        />
        <button>Send</button>
      </form>
    </div>
  );
}

PostComments.defaultProps = {};

PostComments.propTypes = {};

export default PostComments;
