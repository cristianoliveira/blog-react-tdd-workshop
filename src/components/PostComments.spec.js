import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';

import PostComments from './PostComments';

describe('Post Comment', () => {
  it('renders nothing', () => {
    const props = {comments: []};

    const component = shallow(<PostComments {...props} />);

    expect(component.type()).toEqual(null);
  });

  it('renders a comment', () => {
    const props = {
      comments: [{body: 'Awesome'}],
      onCommentFormSubmit: jest.fn(),
      onFocus: jest.fn(),
    };

    const component = shallow(<PostComments {...props} />);

    expect(component.find('Comment')).toHaveLength(1);
  });

  it('renders more than one comment', () => {
    const props = {
      comments: [{body: 'Awesome'}, {body: 'Awesome'}],
      onCommentFormSubmit: jest.fn(),
      onFocus: jest.fn(),
    };

    const component = shallow(<PostComments {...props} />);

    expect(component.find('Comment').length).toBeGreaterThan(1);
  });

  it('renders form', () => {
    const props = {
      comments: [{body: 'Awesome'}, {body: 'Worse'}],
      onCommentFormSubmit: jest.fn(),
      onFocus: jest.fn(),
    };

    const component = shallow(<PostComments {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
