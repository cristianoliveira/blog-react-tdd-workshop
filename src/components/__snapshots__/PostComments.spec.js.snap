// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Comment renders form 1`] = `
<PostComments
  comments={
    Array [
      Object {
        "body": "Awesome",
      },
      Object {
        "body": "Worse",
      },
    ]
  }
  onCommentFormSubmit={[MockFunction]}
  onFocus={[MockFunction]}
>
  <div>
    <div
      className="post-comments"
    >
      <Comment
        body="Awesome"
        key="0"
      >
        <div
          className="post-comment"
        >
          Awesome
        </div>
      </Comment>
      <Comment
        body="Worse"
        key="1"
      >
        <div
          className="post-comment"
        >
          Worse
        </div>
      </Comment>
    </div>
    <form
      className="post-form-comment"
      onSubmit={[MockFunction]}
    >
      <label>
        New commentary
      </label>
      <textarea
        cols="40"
        id="text"
        name="text"
        onFocus={[MockFunction]}
        rows="8"
      />
      <button>
        Send
      </button>
    </form>
  </div>
</PostComments>
`;
